### DISCUSSION TITLE
Toggle BLE Switch with Room Assistant- Switchbot

### DISCUSSION THREAD

**Discussion by TCoughlin02** [mKeRix/room-assistant#662] [Q&A]
I have home assistant installed on a VM and was starting to build out room assistant on a few old pi's that I have.  I recently got a switchbot that i cannot connect to from my VM at the moment.  Could I use room assistant to connect and control the BLE switch, or would I need some other method on the nearest pi?

**Comment by mKeRix** [comment]
At the moment room-assistant does not support Switchbots, although it could in theory. Unfortunately, I don't have a Switchbot device to test against, so I won't be able to write the code for this atm. I'm open for accepting contributions in this regard though. The framework for building this should already be (mostly) there, you could hook into the switch entity type and write a new integration that depends on the `bluetooth` module.
